fe/fe-core/src/main/java/com/starrocks/sql/optimizer/base/HashDistributionSpec.java

```diff
@@ -27,6 +35,33 @@ public boolean isSatisfy(DistributionSpec spec) {
         }
 
         HashDistributionSpec other = (HashDistributionSpec) spec;
+        HashDistributionDesc.SourceType thisSourceType = hashDistributionDesc.getSourceType();
+        HashDistributionDesc.SourceType otherSourceType = other.hashDistributionDesc.getSourceType();
+        // shuffle_local may satisfy shuffle_agg or shuffle_join
```

Comment:
```suggestion
        // shuffle_local satisfy shuffle_agg or shuffle_join
```

Reply:
It's actually "may", because It needs checks below